Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/Utilities/ValidationUtil.cs

```diff
@@ -0,0 +1,305 @@
+                    }
+                }
+
+                return element.GetBindingInfoOfSingleElement(element.DataContext ?? dataItem, dataItem, twoWayOnly, useBlockList);
+            }
+
+            Stack<DependencyObject> children = new Stack<DependencyObject>();
+            Stack<object> dataContexts = new Stack<object>();
+            children.Push(element);
+            dataContexts.Push(element.DataContext ?? dataItem);
+
+            while (children.Count != 0)
+            {
+                bool searchChild = true;
+                DependencyObject child = children.Pop();
+                object inheritedDataContext = dataContexts.Pop();
+                object dataContext = inheritedDataContext;
+
+                // Skip this particular child element if it is one of the excludedTypes
+                if (excludedTypes != null)
+                {
+                    foreach (Type excludedType in excludedTypes)
+                    {
+                        if (excludedType != null && excludedType.IsInstanceOfType(child))
+                        {
+                            searchChild = false;
+                            break;
+                        }
+                    }
+                }
+
+                // Add the bindings of the child element and push its children onto the stack of remaining elements to search
+                if (searchChild)
+                {
+                    FrameworkElement childElement = child as FrameworkElement;
+                    if (childElement != null)
+                    {
+                        dataContext = childElement.DataContext ?? inheritedDataContext;
+                        bindingData.AddRange(childElement.GetBindingInfoOfSingleElement(inheritedDataContext, dataItem, twoWayOnly, useBlockList));
+                    }
+
+                    int childrenCount = VisualTreeHelper.GetChildrenCount(child);
+                    for (int childIndex = 0; childIndex < childrenCount; childIndex++)
+                    {
+                        children.Push(VisualTreeHelper.GetChild(child, childIndex));
+                        dataContexts.Push(dataContext);
+                    }
+                }
+            }
+
+            return bindingData;
+        }
+
+        /// <summary>
+        /// Gets a list of the specified FrameworkElement's DependencyProperties. This method will return all
+        /// DependencyProperties of the element unless 'useBlockList' is true, in which case all bindings on elements
+        /// that are typically not used as input controls will be ignored.
+        /// </summary>
+        /// <param name="element">FrameworkElement of interest</param>
+        /// <param name="useBlockList">If true, ignores elements not typically used for input</param>
+        /// <returns>List of DependencyProperties</returns>
+        public static List<DependencyProperty> GetDependencyProperties(this FrameworkElement element, bool useBlockList)
```

Comment:
Same here?

Reply:
Same. We prefered to have a standalone DataGrid assembly.